internal/clients/aws.go

```diff
@@ -230,18 +99,57 @@ func (m *metaOnlyPrimary) Meta() any {
 	return m.meta
 }
 
-func configureNoForkAWSClient(ctx context.Context, ps *terraform.Setup, config *SetupConfig) error { //nolint:gocyclo
-	p := *config.TerraformProvider
-	diag := p.Configure(context.WithoutCancel(ctx), &tfsdk.ResourceConfig{
-		Config: ps.Configuration,
+// configureNoForkAWSClient populates the supplied *terraform.Setup with
+// Terraform Plugin SDK style AWS client (Meta) and Terraform Plugin Framework
+// style FrameworkProvider
+func configureNoForkAWSClient(ctx context.Context, ps *terraform.Setup, config *SetupConfig, awsCfg *aws.Config, creds aws.Credentials, pc *v1beta1.ProviderConfig) error { //nolint:gocyclo
+	tfAwsConnsCfg := xpprovider.AWSConfig{
+		AccessKey:                     creds.AccessKeyID,
+		EC2MetadataServiceEnableState: imds.ClientDefaultEnableState,
```

Comment:
Do we know the implications of this? A comment would help here. Have we preserved the old implementation?

Reply:
agreed, I can leave a comment here.

The thing is, TF config attribute `skip_metadata_api_check` is implemented as a "nullable boolean" at TF side, because the corresponding AWS config `EC2MetadataServiceEnableState` is a 3-state enum. 
- nil ->`ClientDefaultEnableState`, 
- false ->`ClientEnabled`, 
- true ->`ClientDisabled`

In the XP provider config API , `skip_metadata_api_check` is a regular bool, not a bool pointer. Therefore we cannot capture the 3-state. Would it be OK to change the `skip_metadata_api_check` to a bool pointer?

Reply:
Hi @erhancagirici,
Thank you for the explanation. Because the `skip_metadata_api_check` is an optional field, we should change it to a `bool`  pointer per Kubernetes API conventions (and we should switch to the camel case naming convention). But I suggest we leave this out of scope for this PR, as long as we preserve the old behavior regarding this metadata API check skipping switch before these changes.

We can revisit the ProviderConfig API in future iterations.